osu.Game/Screens/Select/BeatmapDetails.cs

```diff
@@ -299,73 +309,6 @@ public DetailBox()
             }
         }
 
-        private class MetadataSection : Container
-        {
-            private readonly FillFlowContainer textContainer;
-            private TextFlowContainer textFlow;
-
-            public MetadataSection(string title)
-            {
-                RelativeSizeAxes = Axes.X;
-                AutoSizeAxes = Axes.Y;
-
-                InternalChild = textContainer = new FillFlowContainer
-                {
-                    Alpha = 0,
-                    RelativeSizeAxes = Axes.X,
-                    AutoSizeAxes = Axes.Y,
-                    Spacing = new Vector2(spacing / 2),
-                    Children = new Drawable[]
-                    {
-                        new Container
-                        {
-                            RelativeSizeAxes = Axes.X,
-                            AutoSizeAxes = Axes.Y,
-                            Child = new OsuSpriteText
-                            {
-                                Text = title,
-                                Font = OsuFont.GetFont(weight: FontWeight.Bold, size: 14),
-                            },
-                        },
-                    },
-                };
-            }
-
-            public string Text
-            {
-                set
-                {
-                    if (string.IsNullOrEmpty(value))
-                    {
-                        this.FadeOut(transition_duration);
-                        return;
-                    }
-
-                    this.FadeIn(transition_duration);
-
-                    setTextAsync(value);
-                }
-            }
-
-            private void setTextAsync(string text)
```

Comment:
what happened to this? you've completely removed the async text flow that used to be here without any explanation of why.

Reply:
It did not appear to be doing much and made the logic more complicated. In hindsight it improves response time when moving through the menu fast?
I can re add it very easily if you would prefer. In the case that it is, do you think it would be alright if the beatmapsetoverlays was also async? Or should I separate them like they were originally?

Reply:
Removing code that looks like it has purpose without asking or stating a reason is usually a bad sign. This code is making up for the shortcoming that setting text on an already loaded/present drawable will not load the underlying text elements asynchronously.

So you either need to add this back, or replace it with a better mechanisn to avoid aynchronous blocking loads of the text content.